deepaas/cmd/execute.py

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env python -*- coding: utf-8 -*-
+
+# Copyright 2020 Spanish National Research Council (CSIC)
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import magic
+import mimetypes
+import os
+import shutil
+import sys
+
+from oslo_config import cfg
+from oslo_log import log
+
+from deepaas.model import loading
+from deepaas.model.v2.wrapper import UploadedFile
+
+cli_opts = [
+    cfg.StrOpt('input_file',
+               short="i",
+               help="""
+Set local input file to predict.
+"""),
+    cfg.StrOpt('content_type',
+               default='application/json',
+               short='ct',
+               help="""
+Especify the content type of the output file. The selected
+option must be available in the model used.
+(by default application/json).
+"""),
+    cfg.StrOpt('output',
+               short="o",
+               help="""
+Save the result to a local file.
+"""),
+    cfg.BoolOpt('url',
+                short='u',
+                default=False,
+                help="""
+Run as input file an URL.
+If this option is set to True, we can use the URL
+of an image as an input file.
+"""),
+]
+
+CONF = cfg.CONF
+CONF.register_cli_opts(cli_opts)
+
+LOG = log.getLogger(__name__)
+
+# Loading the model installed
+
+
+def model_name():
+    global MODEL_NAME
+    try:
+        for name, model in loading.get_available_models("v2").items():
+            MODEL_NAME = name
+    except Exception as e:
+        LOG.warning("Error loading models: %s", e)
+
+
+def prediction(input_file, file_type, content_type):
+    model_name()  # Function to know the name of the installed model
+
+    package_name = MODEL_NAME
+    predict_data = __import__(package_name).api.predict_data  # Import function
```

Comment:
This assumes that every package has an `api` module, but may not be the case. I would suggest to actually use the entry points, so you rewrite the `model_name()` function to return the model name and the model object, as follows:

    def get_model_name():
        (...)
        return model_name, model

So here you can do:

    model_name, model_obj = get_model()
    predict_data = model_obj.predict_data
    predict_url = model_obj.predict_url